### PR TITLE
Merge dev branch to master

### LINKED ISSUE #1: Fix compile time warning in dev
Currently when I try to compile I am getting lot of warnings incompatible pointers format issue etc,

<Details>

```c
vibhoothiiaanand@derpBox /V/b/m/C-lings (dev)> gcc compile.c
compile.c:25:21: warning: incompatible pointer types passing 'char *[512]' to parameter of type 'char *'
      [-Wincompatible-pointer-types]
            sprintf(directory, dir_list.array[j]);
                    ^~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_stdio.h:47:28: note:
      expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                           ^~~
compile.c:25:32: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
            sprintf(directory, dir_list.array[j]);
                               ^~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_stdio.h:47:56: note:
      expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                                                       ^~~~~~~~~~~
compile.c:25:32: note: treat the string as an argument to avoid this
            sprintf(directory, dir_list.array[j]);
                               ^
                               "%s",
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_stdio.h:47:56: note:
      expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                                                       ^
compile.c:30:25: warning: more '%' conversions than data arguments [-Wformat]
                scanf("%c");
                       ~^
compile.c:95:13: warning: incompatible pointer types passing 'char *[512]' to parameter of type 'char *'
      [-Wincompatible-pointer-types]
    sprintf(gcc,"%s%s","gcc ",paths);
            ^~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_stdio.h:47:28: note:
      expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                           ^~~
compile.c:95:31: warning: format specifies type 'char *' but the argument has type 'char **' [-Wformat]
    sprintf(gcc,"%s%s","gcc ",paths);
                   ~~         ^~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_stdio.h:47:56: note:
      expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                                                       ^~~~~~~~~~~
compile.c:96:20: warning: incompatible pointer types passing 'char *[512]' to parameter of type 'const char *'
      [-Wincompatible-pointer-types]
    int j = system(gcc);
                   ^~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stdlib.h:190:25: note:
      passing argument to parameter here
int      system(const char *) __DARWIN_ALIAS_C(system);
                            ^
compile.c:122:35: warning: incompatible pointer types passing 'char **' to parameter of type 'char *'; dereference with *
      [-Wincompatible-pointer-types]
    sprintf(file_name,"%s",strtok(files,"\n"));
                                  ^~~~~
                                  *
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_stdio.h:47:56: note:
      expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                                                       ^~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/string.h:90:20: note:
      passing argument to parameter '__str' here
char    *strtok(char *__str, const char *__sep);
                      ^
compile.c:122:13: warning: incompatible pointer types passing 'char *[512]' to parameter of type 'char *'
      [-Wincompatible-pointer-types]
    sprintf(file_name,"%s",strtok(files,"\n"));
            ^~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/secure/_stdio.h:47:28: note:
      expanded from macro 'sprintf'
  __builtin___sprintf_chk (str, 0, __darwin_obsz(str), __VA_ARGS__)
                           ^~~
compile.c:123:20: warning: format specifies type 'char *' but the argument has type 'char **' [-Wformat]
    printf("%s \n",files);
            ~~     ^~~~~
compile.c:124:19: warning: incompatible pointer types passing 'char **' to parameter of type 'const char *'; dereference
      with * [-Wincompatible-pointer-types]
   if((fp = fopen(files,"r")) == NULL)
                  ^~~~~
                  *
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stdio.h:153:37: note:
      passing argument to parameter '__filename' here
FILE    *fopen(const char * __restrict __filename, const char * __restrict __mode) __DARWIN_ALIAS_STARTING(__MAC_...
                                       ^
10 warnings generated.
```

<Details>

### LINKED ISSUE #2: clang error during runtime in dev
Currently there are some set of strange symbols which does give the expected output

<Details>

```c
problem-
��:
   /Volumes/buildBox/misc/C-lings/Problems/1.c
��:
   /Volumes/buildBox/misc/C-lings/Problems/2.c
��:
   /Volumes/buildBox/misc/C-lings/Problems/3.c
��:
   /Volumes/buildBox/misc/C-lings/Problems/a.out
clang: error: no such file or directory: '��:
                                             /Volumes/buildBox/misc/C-lings/Problems/1.c'
clang: error: no input files
 not compiled,
 solve error to compile
 for hint type <problem.hint>
```

<Details>

### PR DESCRIPTION
**Pull request by Vishwaak**
The dev branch is rebased properly over the master branch. Fixes issue #1  and #2

### CODE REVIEW

**Review by vibhoothi — CHANGES_REQUESTED** [review]
Please cleanup the commits,
It has lot of noise,

What I would suggest is 
- [x] Soft reset all 7
- [x] Have a single commit which explains the details what you did short and precise.

**Review by vibhoothi — COMMENTED** [review]
Commit messages seems fine but still, it is not able to watch/verify.

<Details>

```c
vibhoothiiaanand@derpBox /V/b/m/C-lings (dev) [255]> ./test verify
problem-
���
/Volumes/buildBox/misc/C-lings/Problems/1.c
���
/Volumes/buildBox/misc/C-lings/Problems/2.c
���
/Volumes/buildBox/misc/C-lings/Problems/3.c
���
/Volumes/buildBox/misc/C-lings/Problems/a.out
clang: error: no such file or directory: '���'
clang: error: no input files
sh: line 1: /Volumes/buildBox/misc/C-lings/Problems/2.c: No such file or directory

problem not compiled,
 solve error to compile
 for hint type <problem.hint>
```
</Details>

**Comment by Vishwaak** [comment]
There seems to be no issue with, watch/verify command in my setup. Also seems to work with clang-10 with no issue.

File: compile.c [hunk 1]
```diff
@@ -3,39 +3,94 @@
...
         {
-            printf("Remove //I AM NOT DONE comment to continue compilation \n");
-            break;
+            sprintf(directory, (char *)dir_list.array[j]);
+            compile(directory);
         }
     }
+    else
+    {
+        printf("Commnad not found \n");
```
Comment (akshayaky) [inline compile.c:55]: Typo here

File: compile.c [hunk 2]
```diff
@@ -3,40 +3,108 @@
...
+    dir_list = get_directory((char *)directory); 
+ 
+    if(strcmp("watch",argv[1]) == 0)
+    {
+        int j=0;
+        while(j<exe_cnt)
+         {
+            sprintf(directory,"%s", dir_list.array[j]);
+            printf("inside while %s", directory);
+	    compile(directory);
```
Comment (akshayaky) [inline compile.c:33]: Not aligned.

File: compile.c [hunk 3]
```diff
@@ -3,40 +3,108 @@
...
+                scanf("%c",u);
+            }
+            else j++;
+            
+        }
+    }
+    else if(strcmp("verify",argv[1]) == 0)
+    {
+        printf("\n Inside verify %s \n",directory);
+	for(int j=1; j<exe_cnt; j++)
```
Comment (akshayaky) [inline compile.c:47]: Not aligned.

File: compile.c [hunk 4]
```diff
@@ -53,7 +121,32 @@ int main()
...
+}
+
+int Complete(char *files)
+{
+   char temp[512] = {0};
+   char *file_name = malloc( 128 * sizeof(char));
+   FILE *fp;
+   char *str = "//I AM NOT DONE";
+    sprintf(file_name,"%s",strtok(files,"\n"));
+    printf("%s \n",files);
```
Comment (akshayaky) [inline compile.c:133]: Not aligned.

File: compile.c [hunk 5]
```diff
@@ -3,40 +3,108 @@
...
+    {
+        printf("problem- \n");
+    }
+    else
+    {
+        perror("getcwd() error");
+    }
+        sprintf(command,"%s%s%s","ls ",cwd,"/Problems/ > file.txt");
+        system(command);
+	printf("\n Total command after strcats: %s \n", command);
```
Comment (akshayaky) [inline compile.c:76]: Not aligned.

File: compile.c [hunk 6]
```diff
@@ -3,40 +3,108 @@
...
+	printf("\n path %s, cwd: %s \n",path,cwd);
+	sprintf(path,"%s%s",cwd,"/Problems/");
+	printf("late path: %s \n", path);
+        int j = 0;
+        while(getline(&line,&len,fp)!= -1){
+                sprintf(result.array[j], "%s%s", path, line);
+		printf("%s",result.array[j]);
+            
+            j++;       
+    }
```
Comment (akshayaky) [inline compile.c:92]: - Uneven indentation.  
 - Better to use a single style of Brace Placement

File: compile.c [hunk 7]
```diff
@@ -53,7 +121,32 @@ int main()
...
+   while(fgets(temp,512,fp) != NULL){
+       if(strstr(temp,str) != NULL)
+       {
+               return(0);
+       }
+   }
+   if(fp)
+    {
+       fclose(fp);
+    }
```
Comment (akshayaky) [inline compile.c:148]: Brace placement is different from the rest.

**Review by akshayaky — CHANGES_REQUESTED** [review]
The lines are not properly indented and different brace placement styles are used throughout the code. Fixing these would improve Code Readability.

**Pull request merged**